Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -107,6 +106,40 @@ void AsyncScrollingCoordinator::handleWheelEventPhase(ScrollingNodeID nodeID, Pl
 }
 #endif
 
+ScrollingStateTree& AsyncScrollingCoordinator::ensureScrollingStateTreeForRootFrameID(FrameIdentifier identifier)
+{
+    auto it = m_scrollingStateTrees.find(identifier);
+    if (it != m_scrollingStateTrees.end())
+        return it->value.get();
+
+    ASSERT_WITH_MESSAGE(!page() || page()->settings().siteIsolationEnabled() || m_scrollingStateTrees.isEmpty(), "With site isolation off, we should only ever have one scoling state tree. Get the frame identifier from LocalFrame::rootFrame.");
+    auto addResult = m_scrollingStateTrees.set(identifier, makeUniqueRef<ScrollingStateTree>(this));
```

Comment:
You can use the `it` you got before to do the insertion. `ensure()` can be a better way to write this.

Reply:
Added use of ensure.